### DISCUSSION TITLE
How to express 'dx' in SymPy

### DISCUSSION THREAD

**Discussion by zzc0430** [sympy/sympy#25206] [Q&A] [answered]
Like: "xdx = ydy"

**Comment by 1e9abhi1e10** [comment]
You can express like:

``` python
x, y = symbols('x y')
dx = symbols('dx')
dy = symbols('dy')

```

**Reply by zzc0430** [reply]
but sympy can't solve $xdx = ydy$ where dx, dy are symbols

**Comment by asmeurer** [comment, marked as answer]
SymPy can't represent differentials directly, but you can represent dy/dx using `Derivative`, like

```py
x = symbols('x')
y = Function('y')
Eq(x, y(x)*Derivative(y(x), x))
```

This equation can be solved with `dsolve`:

```py
>>> ode = Eq(x, y(x)*Derivative(y(x), x))
>>> dsolve(ode, y(x))
[Eq(y(x), -sqrt(C1 + x**2)), Eq(y(x), sqrt(C1 + x**2))]
```

**Reply by zzc0430** [reply]
Are there any plans to add a differential class definition to the SymPy?
